### PR TITLE
x86 IL fixes found during tracetesting

### PR DESCRIPTION
**Pull request by DMaroo**
# DO NOT SQUASH

**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [x] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

Fixed various issues in the x86 IL discovered during tracetesting. It already contains fixes included in #3401, #3402 and #3407 found by @pelijah. To avoid such double-work, I am merging all the fixes I have found till now. This doesn't mean that the IL is correct and verified now, because the tracetesting is not complete yet. But this will definitely fix blatant bugs in the IL.

**Test plan**

Green CI.

**Closing issues**

None.

### CODE REVIEW

**Comment by XVilka** [comment]
@DMaroo how is it going with this one?

**Comment by XVilka** [comment]
@DMaroo could you please finish it before GSoC coding period starts? It's very important, also necessary for other people if they continue to work on RzIL things.

**Comment by DMaroo** [comment]
Hi, sorry for leaving this! I was really busy with other work.

It is merge-able as of now, since the accuracy isn't "bad". But yeah, can be definitely be improved by a bit. I think we should merge whatever changes we have on 1st May. Till then I'll try to improve it as much as I can.

(@XVilka)

**Comment by XVilka** [comment]
@DMaroo please rebase and make it green. Also, if you have any updates - please commit there too.

**Comment by DMaroo** [comment]
@XVilka I will rebase and make it green this weekend, however, the tracetesting isn't entirely complete yet. I will add the rest of the changes later.

**Review by XVilka — CHANGES_REQUESTED** [review]
190 broken tests

File: librz/analysis/arch/x86/x86_il.c [hunk 1]
```diff
@@ -2732,14 +2847,14 @@ IL_LIFTER(popfq) {
...
-RzILOpEffect *x86_push_helper_impl(RzILOpPure *val, unsigned int user_op_size, unsigned int bitness, const X86ILIns *ins) {
-	X86Mem stack_mem;
-	/* The correct register will automatically be chosen if we use RSP */
-	stack_mem.base = X86_REG_RSP;
-	stack_mem.disp = 0;
-	stack_mem.index = X86_REG_INVALID;
-	stack_mem.scale = 1;
-	stack_mem.segment = X86_REG_SS;
+RzILOpEffect *x86_push_helper_impl(RzILOpPure *val, unsigned int user_op_size, unsigned int bitness, const X86ILIns *ins, ut64 pc) {
+	// X86Mem stack_mem;
```
Comment (XVilka) [inline librz/analysis/arch/x86/x86_il.c]: What with this commented code?

File: test/db/asm/x86_16 [hunk 2]
```diff
@@ -5,27 +5,28 @@ ad "aam" d40a 0x0 (seq (set temp_al (cast 8 false (var ax))) (set ax (| (& (var
...
+a "jmp 0x0" ebfe 0x0 (jmp (cast 16 false (bv 16 0x0)))
+a "jmp 0x10" eb0e 0x0 (jmp (cast 16 false (bv 16 0x10)))
+a "jmp 0x34" eb32 0x0 (jmp (cast 16 false (bv 16 0x34)))
+a "jne 0x14" 7512 0x0 (branch (! (var zf)) (jmp (bv 16 0x14)) nop)
+a "jnz 0x94" 660f858d000000 0x0 (branch (! (var zf)) (jmp (bv 16 0x94)) nop)
+a "jnz -0x94" 660f8565ffffff 0x0 (branch (! (var zf)) (jmp (bv 16 0xff6c)) nop)
+a "jno -0x34" 71ca 0x0 (branch (! (var of)) (jmp (bv 16 0xffcc)) nop)
 dB "jmp 0xfec50" e95bec
-d "jmp 0x1fec50" e95bec 0x1ffff2 (jmp (+ (bv 16 0xfff2) (bv 16 0xec50)))
+dB "jmp 0xec50" e95bec 0x1ffff2 (jmp (cast 16 false (bv 16 0xec50)))
```
Comment (XVilka) [inline test/db/asm/x86_16:18]: Why this one is broken?

**Review by XVilka — APPROVED** [review]

File: librz/analysis/arch/x86/x86_il.c (hunk 1 above)
Comment (DMaroo) [reply librz/analysis/arch/x86/x86_il.c]: It's meant to be deleted. Totally missed it, thanks!

File: test/db/asm/x86_16 (hunk 2 above)
Comment (DMaroo) [reply test/db/asm/x86_16:18]: The disassembly doesn't match. `jmp 0xec50` is expected, but we get `jmp 0x1fec50`, which doesn't even make sense for a 16-bit address space.

Comment (XVilka) [reply test/db/asm/x86_16:18]: Right

**Comment by DMaroo** [comment]
Thanks a lot @XVilka for finishing this and checking up on this! x86 IL is now quite mature :tada:

**Pull request merged**